### PR TITLE
fix: Added unique filter to PSRule results & updated action group used for testing

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added unique filter to PSRule results
- Updated action group used for testing (nullable types, location warning)

| Pipeline |
| - |
| [avm.res.insights.action-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.action-group.yml) |

Example of a test where I provoked PSRule to fail:
image
([source](https://github.com/AlexanderSehr/bicep-registry-modules/actions/runs/6596572027))

### CODE REVIEW

File: avm/res/insights/action-group/tests/e2e/max/main.test.bicep [hunk 1]
```diff
@@ -78,5 +80,15 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
       Environment: 'Non-Prod'
       Role: 'DeploymentValidation'
     }
+
```
Comment (eriqua) [inline avm/res/insights/action-group/tests/e2e/max/main.test.bicep:83]: I'd suggest to remove the workaround for anything which is neither defaults nor waf-aligned.

File: avm/res/insights/action-group/tests/e2e/waf-aligned/main.test.bicep [hunk 2]
```diff
@@ -46,37 +47,25 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
...
-        name: 'TestUser_-SMSAction-'
-        phoneNumber: '2345678901'
-      }
-    ]
     tags: {
       'hidden-title': 'This is visible in the resource name'
       Environment: 'Non-Prod'
       Role: 'DeploymentValidation'
     }
+
```
Comment (eriqua) [inline avm/res/insights/action-group/tests/e2e/waf-aligned/main.test.bicep:57]: Please remove empty line
```suggestion
```

File: avm/res/insights/action-group/tests/e2e/max/main.test.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/insights/action-group/tests/e2e/max/main.test.bicep:83]: Removed @eriqua

Comment (AlexanderSehr) [reply avm/res/insights/action-group/tests/e2e/max/main.test.bicep]: done @eriqua

**Review by eriqua — APPROVED** [review]
:ok:

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**